ingestor/shared/src/main/scala/za/co/absa/hyperdrive/shared/utils/SparkUtils.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ *  Copyright 2019 ABSA Group Limited
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package za.co.absa.hyperdrive.shared.utils
+
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.types.{ArrayType, StructField, StructType}
+
+object SparkUtils {
+
+  def setAllColumnsNullable(df: DataFrame): DataFrame = {
+    val newSchema = setSchemaNullability(df.schema, desiredNullability = true)
+    df.sparkSession.createDataFrame(df.rdd, newSchema)
+  }
+
+  def setAllColumnsNonNullable(df: DataFrame): DataFrame = {
```

Comment:
Why did you add this method, because it is unused?

Reply:
It was for completeness, i.e. nullable and non-nullable. On the other hand, it breaks YAGNI. I agree with you and will go for YAGNI for now.